packages/code-snippet-block/src/CodeSnippetBlock.tsx

```diff
@@ -1,53 +1,127 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
-import 'tailwindcss/tailwind.css';
-
-import { FC, ReactElement } from 'react';
-import { debounce } from '@frontify/fondue';
-import { radiusStyleMap, toRgbaString } from '@frontify/guideline-blocks-shared';
+import { Extension } from '@codemirror/state';
 import { useBlockSettings, useEditorState } from '@frontify/app-bridge';
+import { Dropdown, DropdownSize, debounce } from '@frontify/fondue';
+import { radiusStyleMap, toRgbaString } from '@frontify/guideline-blocks-shared';
+import { langs } from '@uiw/codemirror-extensions-langs';
+import * as themes from '@uiw/codemirror-themes-all';
+import CodeMirror from '@uiw/react-codemirror';
+import { ReactElement, useEffect, useState } from 'react';
+import 'tailwindcss/tailwind.css';
+import { CodeSnippetProps, Language, Settings, languageNameMap } from './types';
 
-import { CodeMirrorEditor } from './components';
-import { DEFAULT_THEME_VALUE } from './constants';
-import { CodeMirrorEditorProps, CodeSnippetProps, Settings } from './types';
-
-export const CodeSnippetBlock: FC<CodeSnippetProps> = ({ appBridge }): ReactElement => {
+export const CodeSnippetBlock = ({ appBridge }: CodeSnippetProps): ReactElement => {
     const [blockSettings, setBlockSettings] = useBlockSettings<Settings>(appBridge);
     const isEditing = useEditorState(appBridge);
+    const [contentValue] = useState(blockSettings.content);
+    const [selectedLanguage, setSelectedLanguage] = useState(blockSettings.language ?? 'plain');
+    const extensions = [] as Extension[];
+
+    useEffect(() => {
+        setSelectedLanguage(blockSettings.language ?? 'plain');
+    }, [blockSettings.language]);
 
     const {
-        content,
-        language,
         borderStyle,
         borderWidth,
         borderColor,
         hasBorder = false,
         withHeading = false,
         withRowNumbers = false,
-        theme = DEFAULT_THEME_VALUE,
+        theme = 'default',
     } = blockSettings;
 
+    const getTheme = () => {
+        if (theme === 'default') {
+            return 'light';
+        } else if (Object.keys(themes).includes(theme)) {
+            return themes[theme];
+        }
+        return 'light';
+    };
```

Comment:
Couldn't this be simplified to 
`const getTheme = () => themes[theme] ?? 'light';`

Reply:
`theme` can not be used as an index for `themes`, because `default` does not exist in the `Themes` type, so I had to do it this way.

Reply:
But I changed it to one condition only :)